tests_and_analysis/test/euphonic_test/test_brille_interpolator.py

```diff
@@ -0,0 +1,294 @@
+from multiprocessing import cpu_count
+
+import pytest
+import numpy as np
+
+from euphonic import ForceConstants, QpointPhononModes, ureg
+from tests_and_analysis.test.utils import get_data_path, get_test_qpts
+from tests_and_analysis.test.euphonic_test.test_force_constants import (
+    get_fc)
+from tests_and_analysis.test.euphonic_test.test_qpoint_phonon_modes import (
+    get_qpt_ph_modes_from_json)
+from tests_and_analysis.test.euphonic_test.test_qpoint_frequencies import (
+    get_qpt_freqs, check_qpt_freqs)
+from tests_and_analysis.test.euphonic_test.test_crystal import (
+    get_crystal, check_crystal)
+from tests_and_analysis.test.euphonic_test.test_spectrum1d import (
+    get_spectrum1d, check_spectrum1d)
+
+# Allow tests with brille marker to be collected and
+# deselected if brille isn't installed
+pytestmark = pytest.mark.brille
+try:
+    from brille import BZTrellisQdc, BZMeshQdc, BZNestQdc
+    from euphonic.brille import BrilleInterpolator
+except ModuleNotFoundError:
+    pass
+
+
+def get_brille_grid(grid_file):
+    filepath = get_data_path('brille_grid', grid_file)
+    if 'trellis' in grid_file:
+        return BZTrellisQdc.from_file(filepath)
+    elif 'mesh' in grid_file:
+        return BZMeshQdc.from_file(filepath)
+    else:
+        return BZNestQdc.from_file(filepath)
+
+
+def test_import_without_brille_raises_err(
+        mocker):
+    # Mock import of brille to raise ModuleNotFoundError
+    import builtins
+    from importlib import reload
+    import euphonic.brille
+    real_import = builtins.__import__
+    def mocked_import(name, *args, **kwargs):
+        if name == 'brille':
+            raise ModuleNotFoundError
+        return real_import(name, *args, **kwargs)
+    mocker.patch('builtins.__import__', side_effect=mocked_import)
+    with pytest.raises(ModuleNotFoundError) as mnf_error:
+        reload(euphonic.brille)
```

Comment:
Just curious why you have to import `euphonic.brille` correctly first before patching `__import__` to raise the module error and then check it on reload. Is this both a check that it can be imported correctly as well as that the correct error is raised if Brille cannot be imported?

Reply:
I think I got this pattern from https://github.com/pace-neutrons/Euphonic/blob/master/tests_and_analysis/test/euphonic_test/test_plot.py#L41-L56 and didn't think too much about it...

The original test was added in #89 https://github.com/pace-neutrons/Euphonic/blame/7c8694618a978ebced0e9eba9a9098645025249b/tests_and_analysis/test/euphonic_test/test_plot.py, I think what you suggested is probably the case, there isn't anything obvious there to state otherwise.